docs/design/Non_Admin_Controller_design.md

```diff
@@ -0,0 +1,133 @@
+
+### Implementation details
+- Backup Workflow
+    - **Non-Admin user creates/updates a Non-Admin backup CR:**
+1. **User creates or updates the NonAdminBackup object**: The user creates or updates a NonAdminBackup custom resource object in the Namespace on which the backup will run within the Kubernetes cluster. The `NonAdminBackup` schema has the `backupSpec`, which is the same as `Backup` CR from the `velero.io/v1` apiVersion.
+
+    ```yaml
+    apiVersion: nac.oadp.openshift.io/v1alpha1
+    kind: NonAdminBackup
+    metadata:
+      name: example
+      namespace: user-namespace
+    spec:
+      backupSpec: {}
+    ``` 
+    - **NAB controller reconiles on this NAB CR:** The NonAdminBackup controller continuously reconciles the NonAdminBackup object's desired state with the actual state in the cluster.
+    - **NAB controller validates the NAB CR and then creates a corresponding Velero Backup CR:** When the NonAdminBackup controller detects a new or modified NonAdminBackup object, it creates or updates a corresponding Velero Backup object within the OADP Namespace using the information provided in the `backupSpec` field of the NonAdminBackup object. The resulting Backup object is named as `nab-<namespace>-<hash>`, where the `<namespace>` is the NonAdminBackup namespace and the `<hash>` is computed from the original NonAdminBackup name. The resulting Backup object is labeled and annotated with the following additional metadata:
+
+    ```yaml
+    metadata:
+      annotations:
+        openshift.io/oadp-nab-origin-name: <NonAdminBackup name>
+        openshift.io/oadp-nab-origin-namespace: <NonAdminBackup Namespace>
+        openshift.io/oadp-nab-origin-uuid: <NonAdminBackup UUID>
+      labels:
+        app.kubernetes.io/managed-by: <OADP NonAdminController id>
+        openshift.io/oadp: 'True'
+    ```
+    - **Velero runs Backup**: Velero executes the backup operation based on the configuration specified in the Velero Backup object. Velero updates the status of the Velero Backup object to reflect the outcome of the backup process.
+    - **Velero runs Backup**: Velero executes the backup operation based on the configuration specified in the Velero Backup object. Velero updates the status of the Velero Backup object to reflect the outcome of the backup process.
+    - **Reconcile loop updates NonAdminBackup object Status**: Upon detecting changes in the status of the Velero Backup object, the NonAdminBackup controller's reconciliation loop updates the Status field of the corresponding NonAdminBackup object with the updated status from the Velero Backup object.
+
+![NAB-Backup Workflow Diagram](nab-backup-workflow.jpg)
+![NAB-Controller Backup Details Diagram](Backup-Workflow-Details.jpg)
+- Restore Workflow
+
+
+## Open Questions and Know Limitations
+- Velero command and pod logs
+- Multiple instance of OADP Operator
```

Comment:
please add:
- status regarding the queue of backups pending or running that may be blocking the non-admin backup.
- e.g. There are currently 5 OADP backups queued.

Reply:
Multiple instances of OADP operator can exist, but we must make sure that no more than one of them enable non-admin. If a second DPA adds enableNonAdmin, that should trigger a validation error.